javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java

```diff
@@ -24,6 +24,7 @@
 public class PrettyPrinterConfiguration {
     private boolean printComments = true;
     private String indent = "    ";
+    private Class<? extends PrettyPrintVisitor> visitorClass;
```

Comment:
the previous version also create new instance for each call, did I miss something?

```
 public String print(Node node) {
        final PrettyPrintVisitor visitor = new PrettyPrintVisitor(configuration);
        node.accept(visitor, null);
        return visitor.getSource();
    }
```

And if the visitor is shared among different print call, test cases fails. That‘s why I used reflection.

For case `retainAll()`

    java.lang.AssertionError: 
    Expected :['int a;' REMOVAL in list at 0, 'int c;' REMOVAL in list at 1, 'int e;' REMOVAL in list at 2]
    Actual   :['intint aint;' REMOVAL in list at 0, 'intint aint;intint cint;' REMOVAL in list at 1, 'intint aint;intint cint;intint eint;' REMOVAL in list at 2]

So I'm a little confused here, maybe we can discuss it on gitter